docs/aemm_interruption_testing.md

```diff
@@ -0,0 +1,91 @@
+ State: NodeName:ip-192-168-123-456.us-east-1.compute.internal StartTime:2020-09-15 21:15:43 +0000 UTC EndTime:0001-01-01 00:00:00 +0000 UTC Drained:false PreDrainTask:0x113c8a0 PostDrainTask:<nil>}
+WARNING: ignoring DaemonSet-managed Pods: default/amazon-ec2-metadata-mock-pszj2, kube-system/aws-node-bl2bj, kube-system/aws-node-termination-handler-2pvjr, kube-system/kube-proxy-fct9f
+2020/09/15 21:13:44 Node "ip-192-168-123-456.us-east-1.compute.internal" successfully cordoned and drained.
+2020/09/15 21:13:45 Sending interruption event to the interruption channel
+2020/09/15 21:13:45 Got interruption event from channel...
+```
+
+This isn't a mistake, by default AEMM will respond to any request for metadata with a spot interruption occurring 2 minutes
+later than the request time.\* AWS Node Termination Handler polls for events every 2 seconds by default, so the effect is
+that new interruption events are found and processed every 2 seconds. 
+
+In reality there will only be a single interruption event, and you can mock this by setting the `spot.time` parameter of
+AEMM when installing it. 
+```
+helm install amazon-ec2-metadata-mock amazon-ec2-metadata-mock-1.6.0.tgz \
+  --set aemm.spot.time="2020-09-09T22:40:47Z" \
+  --namespace default
+```
+
+Now when you check the logs you should only see a single event get processed. 
+
+For more ways of configuring AEMM check out the [Helm configuration page](https://github.com/aws/amazon-ec2-metadata-mock/tree/master/helm/amazon-ec2-metadata-mock).
+
+## Node Termination Handler E2E Tests
+
+AEMM started out as a test server for aws-node-termination-handler's end-to-end tests in this repo. We use AEMM throughout
+our end to end tests to create interruption notices.
+
+The e2e tests install aws-node-termination-handler using Helm and set the metadata url [here](https://github.com/aws/aws-node-termination-handler/blob/master/test/e2e/spot-interruption-test#L36).
+This becomes where aws-node-termination-handler looks for metadata; other applications on the node still look at the real
+EC2 metadata service.
+
+We set the metadata url environment variable [here](https://github.com/aws/aws-node-termination-handler/blob/master/test/k8s-local-cluster-test/run-test#L18)
+for the local tests that use a kind cluster, and [here](https://github.com/aws/aws-node-termination-handler/blob/master/test/eks-cluster-test/run-test#L117)
+for the eks-cluster e2e tests.
+
+Check out the [ReadMe](https://github.com/aws/aws-node-termination-handler/tree/master/test) in our test folder for more
+info on the e2e tests. 
+
+---
+
+\* Only the first two unique IPs to request data from AEMM receive spot itn information in the response. This was introduced
+in AEMM v1.6.0. This can be overridden with a configuration paramter. For previous versions there is no unique IP restriction. 
```

Comment:
paramter --> parameter

Reply:
f

Reply:
was about to ask how spellcheck didn't catch it

Reply:
🤖